### DISCUSSION TITLE
cant run sudo

### DISCUSSION THREAD

**Discussion by jimc** [arighi/virtme-ng#36] [Q&A] [answered]
Im a recent convert from good-ole virtme, with some teething pains.
primarily, cant run sudo.

atm, Im still running the kernel config'd for 9p, so --force-9p is necessary.

[jimc@frodo f2]$ uname -a
Linux frodo 6.6.0-f2-virtme-00036-gc2d23a2734fe-dirty #178 SMP PREEMPT_DYNAMIC Sun Nov 19 07:30:27 MST 2023 x86_64 GNU/Linux
[jimc@frodo f2]$ sudo su
bash: /bin/sudo: Permission denied
[jimc@frodo f2]$ ll /bin/sudo
---s--x--x 1 root root 202872 Jul 24 18:00 /bin/sudo
[jimc@frodo f2]$ ll /usr/bin/sudo
---s--x--x 1 root root 202872 Jul 24 18:00 /usr/bin/sudo

since virtme-run got to root, I never encountered this before.

**Comment by jimc** [comment]
but passing --user root  gets me there.

**Comment by arighi** [comment]
hm... that's odd, virtme-ng is supposed to create a custom `/etc/sudoers` (that overrides the host's sudoers) when you run as non-root and add a rule for your `$USER` to execute any command with sudo in a passowrdless way.

You can definitely workaround with `--user root` if you are fine to start your seesion as root, but it'd be interesting to understand why sudo isn't working in your case.

Does it happen with any kernel? For example if you test your host's kernel (`vng -r`) , is sudo still returning a permission denied?

**Reply by jimc** [reply]
from my build-dir:  ./builds/v2

[jimc@frodo v2]$ vng -r
mount: mounting /dev/root on /newroot/ failed: No such file or directory
Failed to mount real root.  We are stuck.

when I 1st installed, I had some trial & error that led me to add sudo.
guessing at errs, I pip installed a few items
I argcomplete was one, there was something about wheel also.

once I got "success", there was some warning issued in the install,
maybe that spoiled the sudoers handling ?

**Reply by jimc** [reply]
[    0.518265] x86/mm: Checked W+X mappings: passed, no W+X pages found.
[    0.518687] Run /init as init process
[    0.519311] virtme initramfs: loading netfs.ko.xz...
[    0.520027] Invalid ELF header magic: != ELF
[    0.535118] virtme initramfs: loading fscache.ko.xz...
[    0.535847] Invalid ELF header magic: != ELF
[    0.550704] FS-Cache: Loaded
[    0.551148] virtme initramfs: loading 9pnet.ko.xz...
[    0.551912] Invalid ELF header magic: != ELF
[    0.564203] 9pnet: Installing 9P2000 support
[    0.564685] virtme initramfs: loading 9p.ko.xz...
[    0.565320] Invalid ELF header magic: != ELF
[    0.573776] 9p: Installing v9fs 9p2000 file system support
[    0.574532] virtme initramfs: loading fuse.ko.xz...
[    0.575521] Invalid ELF header magic: != ELF
[    0.591580] fuse: init (API version 7.38)
[    0.592253] virtme initramfs: loading virtiofs.ko.xz...
[    0.592934] Invalid ELF header magic: != ELF
[    0.597381] virtme initramfs: loading 9pnet_virtio.ko.xz...
[    0.598177] Invalid ELF header magic: != ELF
[    0.600844] virtme initramfs: loading virtio_console.ko.xz...
[    0.601685] Invalid ELF header magic: != ELF
[    0.607031] virtme initramfs: loading overlay.ko.xz...
[    0.607702] Invalid ELF header magic: != ELF
[    0.621951] virtme initramfs: mounting hostfs...
[    0.622657] virtio-fs: tag <ROOTFS> not found
[    0.623386] 9pnet_virtio: no channels available for device /dev/root
mount: mounting /dev/root on /newroot/ failed: No such file or directory
Failed to mount real root.  We are stuck.
[    5.624681] Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000100
[    5.625285] CPU: 0 PID: 1 Comm: init Not tainted 6.5.11-300.fc39.x86_64 #1
[    5.625789] Hardware name: Bochs Bochs, BIOS Bochs 01/01/2011
[    5.626151] Call Trace:
[    5.626314]  <TASK>
[    5.626458]  dump_stack_lvl+0x47/0x60
[    5.626708]  panic+0x193/0x350
[    5.626928]  do_exit+0x990/0xb10
[    5.627133]  ? srso_alias_return_thunk+0x5/0x7f
[    5.627412]  ? cgroup_rstat_updated+0x46/0xc0
[    5.627687]  ? srso_alias_return_thunk+0x5/0x7f
[    5.627966]  do_group_exit+0x31/0x80
[    5.628188]  __x64_sys_exit_group+0x18/0x20
[    5.628444]  do_syscall_64+0x60/0x90
[    5.628672]  ? srso_alias_return_thunk+0x5/0x7f
[    5.628950]  ? exc_page_fault+0x7f/0x180
[    5.629191]  entry_SYSCALL_64_after_hwframe+0x6e/0xd8
[    5.629499] RIP: 0033:0x4256be
[    5.629699] Code: dc 12 00 48 8b 35 0a dc 12 00 48 8b 05 1b dc 12 00 e9 7c fe ff ff 66 0f 1f 44 00 00 f3 0f 1e fa 48 63 ff b8 e7 00 00 00 0f 05 <ba> 3c 00 00 00 0f 1f 44 00 00 48 89 d0 0f 05 eb f9 90 55 48 89 e5
[    5.630805] RSP: 002b:00007ffc00461048 EFLAGS: 00000202 ORIG_RAX: 00000000000000e7
[    5.631265] RAX: ffffffffffffffda RBX: 0000000000000004 RCX: 00000000004256be
[    5.631699] RDX: 00007f6946491030 RSI: 0000000000000000 RDI: 0000000000000001
[    5.632138] RBP: 00007ffc00461050 R08: 0000000000000000 R09: 00007f694648d609
[    5.632697] R10: 00000000005524e0 R11: 0000000000000202 R12: 00007ffc00461e82
[    5.633247] R13: 0000000000436e61 R14: 00007ffc00461320 R15: 0000000000000000
[    5.633806]  </TASK>
[    5.634174] Kernel Offset: 0x13000000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[    5.635008] ---[ end Kernel panic - not syncing: Attempted to kill init! exitcode=0x00000100 ]---

**Comment by arighi** [comment]
> from my build-dir: ./builds/v2
> 
> [jimc@frodo v2]$ vng -r mount: mounting /dev/root on /newroot/ failed: No such file or directory Failed to mount real root. We are stuck.
> 
> when I 1st installed, I had some trial & error that led me to add sudo. guessing at errs, I pip installed a few items I argcomplete was one, there was something about wheel also.
> 
> once I got "success", there was some warning issued in the install, maybe that spoiled the sudoers handling ?

What distro / kernel are you using? What do you see with `vng -r --verbose`?

**Comment by arighi** [comment]
ok nevermind I guess the oops is what you see with verbose mode. It looks like the problem is xz compressed modules, we should have fixed that with 918afa5 ("mkinitramfs: support zstd compressed kernel modules"), which version of virtme-ng are you using?

**Comment by arighi** [comment, marked as answer]
Any update? I think this one is fixed in the recent versions. Closing for now.
